### ISSUE TITLE
ffmpeg failure

### ISSUE THREAD

**Issue by f1vefour** [warren-bank/node-hls-downloader-tubitv#9]
I'm getting this error after a download completes.

```
download complete.

starting ffmpeg conversion of HLS stream to mp4 file..
ffmpeg error:
Error: ffmpeg error: media file not found
    at /usr/local/lib/node_modules/@warren-bank/node-hls-downloader-tubitv/lib/fallback.js:35:14
    at new Promise (<anonymous>)
    at run_fallback_ffmpeg_video_conversion (/usr/local/lib/node_modules/@warren-bank/node-hls-downloader-tubitv/lib/fallback.js:12:10)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async process_html (/usr/local/lib/node_modules/@warren-bank/node-hls-downloader-tubitv/lib/process_cli.js:105:9)
    at async process_url (/usr/local/lib/node_modules/@warren-bank/node-hls-downloader-tubitv/lib/process_cli.js:67:5)
    at async process_argv (/usr/local/lib/node_modules/@warren-bank/node-hls-downloader-tubitv/lib/process_cli.js:379:7)
```
ffmpeg (v4.4.2) is in my path and working

**Comment by warren-bank** [comment]
yeah.. these comments in an older issue explain the problem:
1. [byte ranges, rather than video segments](https://github.com/warren-bank/node-hls-downloader-tubitv/issues/8#issuecomment-1562572204)
2. [ffmpeg workaround](https://github.com/warren-bank/node-hls-downloader-tubitv/issues/8#issuecomment-1562724866)
3. [v1.5.4](https://github.com/warren-bank/node-hls-downloader-tubitv/issues/8#issuecomment-1563554475)

in short, strictly speaking, tubitv isn't hosting HLS video any longer..
and ffmpeg doesn't like it when HLS manifests only contain a single media segment

**Comment by warren-bank** [comment]
quick test:

```bash
  tubidl --version

  # 1.5.4
```

* [series: Mr. Bean The Animated Series](https://tubitv.com/series/2176/mr-bean-the-animated-series)
  - each episode is 12 minutes
* [episode: S01E01 - In the Wild](https://tubitv.com/tv-shows/381933/s01-e01-in-the-wild)

```bash
  set tmpdir="e:\.workspace"
  mkdir %tmpdir%
  cd /D %tmpdir%

  tubidl -q -mc 5 -u "https://tubitv.com/tv-shows/381933/s01-e01-in-the-wild"

  # starting download of 1 video data files..
  # done
  # 
  # starting download of 1 audio data files..
  # done
  # 
  # download complete.
  # 
  # starting ffmpeg conversion of HLS stream to mp4 file..
  # ffmpeg error:
  # starting fallback ffmpeg conversion to mp4 file..
  # done
```

```bash
  ffplay "Mr. Bean The Animated Series/S01E01 - In the Wild/mp4/video.mp4"
```

conclusions:

* the ffmpeg workaround added in v1.5.4 still seems to work

**Comment by warren-bank** [comment]
actually..
taking a closer look at your error message..
it seems that:
* you are attempting to run the ffmpeg workaround: `run_fallback_ffmpeg_video_conversion`
* and it's throwing an error because one of the [two files that it assumes to exist](https://github.com/warren-bank/node-hls-downloader-tubitv/blob/v1.5.4/lib/fallback.js#L23-L30) do not

question:
* what are the filenames of the video and audio files that did download (to their respective directories)?

**Comment by f1vefour** [comment]
```
rooted@odroid:~$ exa --tree Doomsday\ Brothers/         Doomsday Brothers
└── S01E01 - Daughter of Humunga
   ├── hls
   │  ├── audio
   │  │  ├── english
   │  │  │  └── segment_0.aac
   │  │  ├── english - audio description
   │  │  │  └── segment_0.aac
   │  │  ├── english - audio description.m3u8
   │  │  └── english.m3u8
   │  ├── master.m3u8
   │  ├── video
   │  │  └── segment_0.ts
   │  └── video.m3u8
   └── mp4
      └── video.english.srt
```

**Comment by warren-bank** [comment]
the problem was that the fallback strategy assumed that the audio stream would be named `stream_0`,
which is probably the case when the media only contains a single audio stream,
but your example has 2: `english` and `english - audio description`

I just pushed [v1.5.6](https://github.com/warren-bank/node-hls-downloader-tubitv/releases/tag/v1.5.6)..
which first tries to find the audio stream having one of the two names: `stream_0` or `english`..
and if neither is found, it then walks the audio directory to find a suitable file..
and only exits with an error message when none can be found.

**Comment by f1vefour** [comment]
It's working now, ffmpeg falls back and joins the audio/video into mp4 correctly.

Thank you for the quick fix, closing the issue.
